File: src/main/java-graphql/io/github/graphql/java/model/RegistryPackageVersionStatisticsResponseProjection.java
package io.github.graphql.java.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;
import java.util.Objects;

/**
 * Response projection for RegistryPackageVersionStatistics
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-07T19:46:14+0800"
)
public class RegistryPackageVersionStatisticsResponseProjection extends GraphQLResponseProjection {

    public RegistryPackageVersionStatisticsResponseProjection() {
    }

    @Override
    public RegistryPackageVersionStatisticsResponseProjection all$() {
        return all$(3);
    }

    @Override
    public RegistryPackageVersionStatisticsResponseProjection all$(int maxDepth) {
        this.downloadsThisMonth();
        this.downloadsThisWeek();
        this.downloadsThisYear();
        this.downloadsToday();
        this.downloadsTotalCount();
        this.typename();
        return this;
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsThisMonth() {
        return downloadsThisMonth(null);
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsThisMonth(String alias) {
        fields.add(new GraphQLResponseField("downloadsThisMonth").alias(alias));
        return this;
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsThisWeek() {
        return downloadsThisWeek(null);
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsThisWeek(String alias) {
        fields.add(new GraphQLResponseField("downloadsThisWeek").alias(alias));
        return this;
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsThisYear() {
        return downloadsThisYear(null);
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsThisYear(String alias) {
        fields.add(new GraphQLResponseField("downloadsThisYear").alias(alias));
        return this;
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsToday() {
        return downloadsToday(null);
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsToday(String alias) {
        fields.add(new GraphQLResponseField("downloadsToday").alias(alias));
        return this;
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsTotalCount() {
        return downloadsTotalCount(null);
    }

    public RegistryPackageVersionStatisticsResponseProjection downloadsTotalCount(String alias) {
        fields.add(new GraphQLResponseField("downloadsTotalCount").alias(alias));
        return this;
    }

    public RegistryPackageVersionStatisticsResponseProjection typename() {
        return typename(null);
    }

    public RegistryPackageVersionStatisticsResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        final RegistryPackageVersionStatisticsResponseProjection that = (RegistryPackageVersionStatisticsResponseProjection) obj;
        return Objects.equals(fields, that.fields);
    }

    @Override
    public int hashCode() {
        return Objects.hash(fields);
    }

}

File: src/main/scala-graphql/io/github/graphql/scala/model/RegistryPackageVersionStatisticsResponseProjection.scala
package io.github.graphql.scala.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection
import java.util.Objects

/**
 * Response projection for RegistryPackageVersionStatistics
 */
@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-07T19:49:36+0800"
)
class RegistryPackageVersionStatisticsResponseProjection extends GraphQLResponseProjection {

    override def all$(): RegistryPackageVersionStatisticsResponseProjection = all$(3)

    override def all$(maxDepth: Int): RegistryPackageVersionStatisticsResponseProjection = {
        this.downloadsThisMonth()
        this.downloadsThisWeek()
        this.downloadsThisYear()
        this.downloadsToday()
        this.downloadsTotalCount()
        this.typename()
        this
    }

    def downloadsThisMonth(): RegistryPackageVersionStatisticsResponseProjection = {
        downloadsThisMonth(null.asInstanceOf[String])
    }

    def downloadsThisMonth(alias: String): RegistryPackageVersionStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsThisMonth").alias(alias))
        this
    }

    def downloadsThisWeek(): RegistryPackageVersionStatisticsResponseProjection = {
        downloadsThisWeek(null.asInstanceOf[String])
    }

    def downloadsThisWeek(alias: String): RegistryPackageVersionStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsThisWeek").alias(alias))
        this
    }

    def downloadsThisYear(): RegistryPackageVersionStatisticsResponseProjection = {
        downloadsThisYear(null.asInstanceOf[String])
    }

    def downloadsThisYear(alias: String): RegistryPackageVersionStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsThisYear").alias(alias))
        this
    }

    def downloadsToday(): RegistryPackageVersionStatisticsResponseProjection = {
        downloadsToday(null.asInstanceOf[String])
    }

    def downloadsToday(alias: String): RegistryPackageVersionStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsToday").alias(alias))
        this
    }

    def downloadsTotalCount(): RegistryPackageVersionStatisticsResponseProjection = {
        downloadsTotalCount(null.asInstanceOf[String])
    }

    def downloadsTotalCount(alias: String): RegistryPackageVersionStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("downloadsTotalCount").alias(alias))
        this
    }

    def typename(): RegistryPackageVersionStatisticsResponseProjection = {
        typename(null.asInstanceOf[String])
    }

    def typename(alias: String): RegistryPackageVersionStatisticsResponseProjection = {
        fields.add(new GraphQLResponseField("__typename").alias(alias))
        this
    }

    override def equals(obj: Any): Boolean = {
        if (this == obj) {
            return true
        }
        if (obj == null || getClass != obj.getClass) {
            return false
        }
        val that = obj.asInstanceOf[RegistryPackageVersionStatisticsResponseProjection]
        Objects.equals(fields, that.fields)
    }

    override def hashCode(): Int = Objects.hash(fields)

}
